### DISCUSSION TITLE
the darker song select bottom bar makes it feel much more claustrophobic

### DISCUSSION THREAD

**Discussion by tovc** [ppy/osu#30678] [Ideas]
![osu_2024-11-16_10-02-01](https://github.com/user-attachments/assets/541e05bc-6650-41a9-be5d-3da1d96d4f04)

I understand that this change was made as somewhat of a stepping stone towards the song select redesign, and in that vein, I'll say that in practice, this isn't something that I want to see in that song select redesign. The bottom bar is ultimately far less important than being able to see more maps, and this issue is exacerbated on smaller screens, and on mobile.

![Screenshot_2024-11-16-10-11-12-145_sh ppy osulazer](https://github.com/user-attachments/assets/48373032-bd21-4394-a06d-4e6c65aa3aa8)

I think that the 'safest' design here is something that priorities free space, rather than one that tries to fill it with endless stacked toolbars.

**Comment by peppy** [comment]
I see you watched the video 🤣

**Reply by tovc** [reply]
I watched it yesterday and fell asleep shortly after. Would I be correct in remembering you saying that you were looking for feedback on it?

**Comment by cl8n** [comment]
I have mixed feelings about this on desktop but wow that mobile view is really bad. will definitely bring this up as a priority to address for next song select

**Reply by tovc** [reply]
In fairness, I am using the largest UI scale available on mobile, but I also wouldn't really want to use anything smaller. I don't love the way it looks on desktop, it just doesn't feel like it's adding anything.

This is somewhat unrelated to the original point, but I think that the search/filter options could do with being hidden in a dropdown to further reclaim some space for the song carousel on mobile.

**Comment by vatei** [comment]
I don't personally mind it on pc, to me it feels less floaty and more like pieces of UI I can interact with, feels more "functional" if that makes any sense.
I will say it looks pretty meh on mobile though

**Reply by tovc** [reply]
I don't think I see what you mean in that first point. It isn't something I can interact with and it doesn't feel like it to me. I guess it's good to have a contrasting opinion on the matter tho.

**Comment by tovc** [comment]
Just adding a few more comments regarding the song select screen, now that I've been paying more attention to it

The song select carousel shouldn't center on a specific difficulty like this:
![osu_2024-11-17_17-32-54](https://github.com/user-attachments/assets/d125aaa1-cb9c-45f1-8222-1a2cc48e37c6)

Instead, it makes a lot more sense to center on the whole mapset, like this:
![osu_2024-11-17_17-33-06](https://github.com/user-attachments/assets/404edc21-f80c-4b42-8df9-ccc8268bea2f)

Having 6 mapsets visible below and 0 mapsets visible above, is just not sensible in my opinion. I can understand how that would cause issues with extremely large mapsets, and something would need to be done there (perhaps limit the distance that the selected mapset can be from the center of the screen?), but I think that the current layout is just silly.
![osu_2024-11-17_17-36-50](https://github.com/user-attachments/assets/9ae925ef-c31d-46f3-b7a2-0fe810fd8d52)

**Reply by tovc** [reply]
So I originally just wanted to edit the spacing between beatmaps, but I ended up just changing a couple of other song select elements to things that I think are a lot cleaner. The footer has been hidden, the song filters have been hidden and the spacing between beatmaps in a single set are gone. I personally think this is much more usable. There are still issues with layering in the song carousel, notably with the selected beatmap/set appearing other items under the carousel, but that seems fixable.

Obviously, the beatmap filters can't be permanently hidden, but I think that having them be in some kind of dropdown would work fine, while also reclaiming a lot of screen space.
![osu_2024-11-17_20-08-05](https://github.com/user-attachments/assets/8b39b2ba-3bbb-4299-aeff-803ed46077fa)

Notably, these changes reclaim a LOT of space on larger UI scales, such as on mobile, which I think makes osu! a lot more usable there.
![osu_2024-11-17_20-14-19](https://github.com/user-attachments/assets/024d99a9-a7d1-4233-85c5-879ff056bb88)

I understand that most of this is only tangentially relevant, as there is a new song select screen in the works, but I figure these design principles are relevant there aswel.
